lib/platform.h

```diff
@@ -85,14 +85,16 @@ namespace cppcheck {
         char defaultSign;  // unsigned:'u', signed:'s', unknown:'\0'
 
         enum PlatformType {
-            Unspecified, // No platform specified
+            MIN_PLATFORM_TYPE,  //used for 'if' condition checks
+            Unspecified = MIN_PLATFORM_TYPE, // No platform specified
             Native, // whatever system this code was compiled on
             Win32A,
             Win32W,
             Win64,
             Unix32,
             Unix64,
-            PlatformFile
+            PlatformFile,
```

Comment:
Imagine that a "Unix128" constant is added. Then there will be undefined behavior in `platformString` because it doesn't return. I am not strongly against this... Because there will be a Wswitch warning and that will eventually be fixed (but possibly after several days).

Reply:
The change then becomes...
```
PlatformFile,
SomeNewUnixType,
MAX_PLATFORM_TYPE=SomeNewUnixType
```

`tostring` functions will report "unknown". But internal error will be thrown. The error message being the impetus to make changes to resolve quickly.

Reply:
If we add it, it should be added after unix64. It's logical to put the unix platforms together.

Reply:
Code changes would be equally as simple. Change the enum and update to/from string.
```
...
    Unix64,
    Unix128,
    PlatformFile,
    MAX_PLATFORM_TYPE=MAX_PLATFORM_TYPE
};
```